pkg/paymasterclient/client.go

```diff
@@ -31,39 +31,68 @@ type Client interface {
 }
 
 type client struct {
-	c *rpc.Client
+	userClient    *rpc.Client
+	sponsorClient *rpc.Client
 }
 
-func New(ctx context.Context, url string, options ...rpc.ClientOption) (Client, error) {
-	c, err := rpc.DialOptions(ctx, url, options...)
+func New(ctx context.Context, userURL, sponsorURL string, options ...rpc.ClientOption) (Client, error) {
```

Comment:
`sponsorURL` can be optional?

Reply:
The sponsorURL cannot be optional. According to the private policy, it must be used in the form of `"https://open-platform-ap.nodereal.io/xxxx/megafuel-testnet"`. This URL is required when users make requests. If a user tries to call the private policy without a sponsor client, it will result in an error.

Reply:
fixed. Make the sponsorUrl optional, so that private policy UUID can only be sent if the user provides this URL. Add a comment explaining the purpose of sponsorUrl and Options.